### PR TITLE
Blazor package lock files

### PR DESCRIPTION
**Pull request by atmgrifter00**
# Pull Request

## 🤨 Rationale

- #1772

## 👩‍💻 Implementation

Following instructions from [MS docs](https://learn.microsoft.com/en-us/nuget/consume-packages/package-references-in-project-files#enabling-the-lock-file).

See [this](https://github.com/dotnet/reproducible-builds/blob/main/Documentation/Reproducible-MSBuild/Techniques/DisableImplicitNuGetFallbackFolder.md) regarding the addition of `DisableImplicitNuGetFallbackFolder` to the projects.

## 🧪 Testing

Verified the pipeline passed with no errors/warnings.

## ✅ Checklist

- [x] I have updated the project documentation to reflect my changes or determined no changes are needed.

### CODE REVIEW

File: packages/nimble-blazor/Examples/Demo.Client/Demo.Client.csproj [hunk 1]
```diff
@@ -4,8 +4,13 @@
     <TargetFramework>net6.0</TargetFramework>
     <Nullable>enable</Nullable>
     <ImplicitUsings>enable</ImplicitUsings>
+	<RestorePackagesWithLockFile>true</RestorePackagesWithLockFile>
```
Comment (msmithNI) [inline packages/nimble-blazor/Examples/Demo.Client/Demo.Client.csproj:7]: Can you do another pass over the .csproj files to clean up the indentation (tab vs space) issues on these new properties? It looks like all the csproj changes (except for NimbleBlazor.csproj) have tabs on the new parts.

**Comment by msmithNI** [comment]
Do we know how these changes will work in conjunction with Renovate updating our .NET dependencies? Do we expect that Renovate will still be able to create PRs for updates (and also handle updating these lock files)?

**Comment by atmgrifter00** [comment]
> Do we know how these changes will work in conjunction with Renovate updating our .NET dependencies? Do we expect that Renovate will still be able to create PRs for updates (and also handle updating these lock files)?

I can't say that I know.

**Review by msmithNI — APPROVED** [review]

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -45,14 +45,18 @@ jobs:
...
       - uses: actions/setup-node@v4
         with:
           node-version: '20'
           registry-url: 'https://registry.npmjs.org'
           cache: 'npm'
       - run: npm ci
+      - run: dotnet restore --locked-mode
+        working-directory: packages/nimble-blazor/
+      - run: dotnet --info
+        working-directory: packages/nimble-blazor/
```
Comment (rajsite) [inline .github/workflows/main.yml:59]: Is this desired or leftover from debugging?

If it's useful to keep around lets just move it down so it's not groupeed under the `Install dependencies` comment

```suggestion

      # Print resolved dotnet info
      - run: dotnet --info
        working-directory: packages/nimble-blazor/
```

**Review by rajsite — APPROVED** [review]

File: .github/workflows/main.yml (hunk 2 above)
Comment (atmgrifter00) [reply .github/workflows/main.yml:59]: Well, it was intentionally introduced as part of [this PR](https://github.com/ni/nimble/pull/1283), with the reasoning that it should make debugging issues related to versioning easier.

**Pull request merged**